scripts/calendar/islamic_generator.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+#  python-holidays
+#  ---------------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from pathlib import Path
+
+from hijridate import convert
+from hijridate.ummalqura import HIJRI_RANGE
+
+class_name = "_IslamicLunar"
+out_file_name = "islamic_dates.py"
+
+class_template = """class {class_name}:
+{holiday_arrays}"""
+
+holiday_array_template = """    {hol_name}_DATES = {{
+{year_dates}
+    }}
+"""
+
+year_template = "        {year}: ({dates}),"
+
+islamic_holidays = (
+    # AL, AZ, BH, BA, BI, DJ, EG, ET, IN, ID, KG, MY, MA, NG, MK, PK, PH,
+    # SA, SG, ES, TN, TR, AE, UZ
+    (10, 1, "EID_AL_FITR"),  # Eid al-Fitr
+    # AL, AZ, BH, BA, BI, DJ, EG, ET, IN, ID, KZ, KG, MY, MA, NG, PK, PH,
+    # SA, SG, ES, TN, TR, AE, UZ
+    (12, 10, "EID_AL_ADHA"),  # Eid al-Adha
+    # BH, DJ, EG, ET, IN, ID, MY, MA, NG, PK, TN, AE
+    (3, 12, "MAWLID"),  # Birthday of Prophet
+    # BH, DJ, EG, ID, MY, MA, TN, AE
+    (1, 1, "HIJRI_NEW_YEAR"),  # Islamic New Year
+    # BH (days-1), IN, PK
+    (1, 10, "ASHURA"),  # Ashura
+    # DJ, ID, MY, AE
+    (7, 27, "ISRA_AND_MIRAJ"),  # Isra and Mi'raj
+    # MY
+    (9, 1, "RAMADAN_BEGINNING"),  # Beginning of Ramadan
+    # MY
+    (9, 17, "NUZUL_AL_QURAN"),  # Nuzul Al-Quran Day
+    # MY
+    (2, 6, "HARI_HOL_JOHOR"),  # Hari Hol of Johor
+)
+
+
+def makelist():
+    h_year_min, h_year_max = (d[0] for d in HIJRI_RANGE)
+
+    dates_array = {}
```

Comment:
I'd prefer this to be less ambiguous (array vs dict):
```suggestion
    dates = {}
```